adaptive_ai/tests/test_complexity/test.py

```
@@ -0,0 +1,304 @@
+    )
+    plt.title("Task Type Distribution")
+    plt.tight_layout()
+    plt.savefig(
+        visualizations_dir / "task_type_distribution.png", dpi=300, bbox_inches="tight"
+    )
+    plt.close()
+
+    # 4. Complexity scores by task type
+    plt.figure(figsize=(12, 6))
+    task_complexity_data = []
+    for task_type, count in results["task_type_counts"].items():
```

Ruff (0.11.9)

Loop control variable `count` not used within loop body

(B007)